packages/admin/resources/views/components/layouts/app/sidebar/index.blade.php

```diff
@@ -49,15 +49,31 @@ class="block w-full text-center"
         <x-filament::layouts.app.sidebar.start />
         {{ \Filament\Facades\Filament::renderHook('sidebar.start') }}
 
+        @php
+            $navigation = \Filament\Facades\Filament::getNavigation();
+
+            $collapsedNavigationGroupLabels = collect($navigation)
+                ->filter(fn (\Filament\Navigation\NavigationGroup $group): bool => $group->isCollapsed())
+                ->map(fn (\Filament\Navigation\NavigationGroup $group): string => $group->getLabel())
+                ->values();
+        @endphp
+
+        <script>
+            if (localStorage.getItem('collapsedGroups') === null) {
+                localStorage.setItem('collapsedGroups', JSON.stringify(@js($collapsedNavigationGroupLabels)))
```

Comment:
What's the `JSON.stringify()` for in addition to `@js()`?

Reply:
Because `@js` passes it to JS as an object, it needs to be stringified as well to be stored in the session